spec/models/site_setting_spec.rb

```diff
@@ -69,6 +69,40 @@
     end
   end
 
+  describe "min_redirected_to_top_period" do
+
+    context "has_enough_top_topics" do
+
+      SiteSetting.topics_per_period_in_top_page = 2
+      SiteSetting.top_page_default_timeframe = 'daily'
+
+      before do
+        SiteSetting.expects(:min_redirected_to_top_period).returns(:daily)
+      end
+
+      it "should_return_a_time_period" do
+        expect(SiteSetting.min_redirected_to_top_period(1.days.ago)).not_to eq(nil)
```

Comment:
That is sort of my point. I have to somehow create Topics that will be returned by TopTopic, but all of my attempts failed, it always returned 0 for the count. I've tried setting it up similar to how the TopTopic specs do it, but for whatever reason it doesn't work here. Any ideas on how I can get TopTopic.where.count to return a specific value?